machines/vcItem.ts

```diff
@@ -1267,13 +1267,14 @@ export const createVcItemMachine = (
   serviceRefs: AppServices,
   vcKey: string
 ) => {
-  const [, idType, id, requestId] = vcKey.split(':');
+  const [, idType, id, requestId, isPinned] = vcKey.split(':');
   return vcItemMachine.withContext({
     ...vcItemMachine.context,
     serviceRefs,
     id,
     idType: idType as VcIdType,
     requestId,
+    isPinned: isPinned as unknown as boolean,
```

Comment:
@tilak-puli Can you suggest if this is the good typescript way of casting `undefined` to a boolean `false` value?

Reply:
@srikanth716 did we try `!!isPinned` to get boolean value?

Reply:
`isPinned` value is getting here in string format since we are storing vc in string format when we do `vcKey.split(':');` it is returning the value in string format, as we cannot assign the string to boolen I used the quick fix. Fixing this by assigning the boolean value directly `isPinned == 'true' ? true : false,`

Reply:
@srikanth716 Can you try out the suggestion from Swati? I tried the below scenarios in the node terminal.

```javascript
# undefined case
> let vcKey = "abcd:bcde:efgh:hjkl"
undefined
# TODO: try the same with 'abcd:bcde:efgh:hjkl:false' , 'abcd:bcde:efgh:hjkl:true', 'abcd:bcde:efgh:hjkl:randomstring'
 
> const [, idType, id, requestId, isPinned] = vcKey.split(':');
undefined
> isPinned
undefined
> !!isPinned
# TODO: see the outputs in different scenarios
```

Reply:
@vharsh @srikanth716 

1. isPinned == 'true' ? true : false

As type of isPinned is boolean. There is a high chance that someone can set value of isPinned to true(boolean type). As `true == 'true'` is false, this will cause a bug

2. !!isPinned
As `!!"false"` is true. which is again inconsistent and causes a bug.

Please try using Boolean wrapper for converting string boolean to boolean type as shown in this blog
https://www.scaler.com/topics/string-to-boolean-javascript/

Reply:
right now it is a string. but as isPinned name seems like a boolean, there is chance that someone might set it to boolean true. which will start failing. so please avoid using string as boolean.

Reply:
We can go ahead for now. but please refactor this whenever you get chance. @srikanth716